Aligns branch protection rules with lagom/lagom

octonato opened the pull request:
This won't be applied for now because probot/settings is broken, but once they fix it it will be applied and we need to have it correctly.

octonato committed:
aligns branch protection rules with lagom/lagom

TimMoore commented on .github/settings.yml:
```diff
@@ -23,15 +23,15 @@ teams:
-        # Required. The list of status checks to require in order to merge into this branch
-        contexts: ["Travis CI - Pull Request", "typesafe-cla-validator"]
+        # The list of status checks to require in order to merge into this branch
+        contexts: ["Travis CI - Branch ", "Travis CI - Pull Request", "typesafe-cla-validator"]
```
```suggestion
        contexts: ["Travis CI - Pull Request", "typesafe-cla-validator"]
```

This shouldn't include "Travis CI - Branch". It would prevent any pull requests from forks, and also Travis is configured to skip branch builds other than on 1.5.x and 1.6.x:

https://github.com/lagom/lagom-samples/blob/e6038105f051afa4f90b8de91f49b381285e9fc2/.travis.yml#L3-L5

octonato committed:
include missing branches.protection settings

All branch protection settings must be included, otherwise, Probot won't apply them.

Also applies @TimMoore code review suggestion.

TimMoore reviewed (commented):
Thanks! Leaving this unapproved so that someone else on the team can have a look.
